cli/src/pcluster/models/cluster_config.py

```diff
@@ -290,6 +290,14 @@ def __init__(
         self.drive_cache_type = Resource.init_param(drive_cache_type)
         self.fsx_storage_type = Resource.init_param(fsx_storage_type)
 
+    def get_drive_cache_type(self):
+        """Drive cache type must be set for HDD (Either "NONE" or "READ"), and must not be set for SDD (None)."""
+        if self.fsx_storage_type == "HDD":
+            if self.drive_cache_type:
+                return self.drive_cache_type
+            return "NONE"
```

Comment:
This is a CFN specific logic. I mean "NONE" is a value accepted by CFN/CDK it's not a meaningful value for the model.
We should move this logic in the cluster_stack.py

Then let's be sure to have a validator that accepts a value for self.drive_cache_type only if fsx_storage_type is HDD

Reply:
Moved. The validation was already in place